core/state_transition.go

```diff
@@ -241,6 +248,9 @@ func (st *StateTransition) to() common.Address {
 }
 
 func (st *StateTransition) buyGas() error {
+	if err := st.applyMetaTransaction(); err != nil {
+		return err
+	}
 	mgval := new(big.Int).SetUint64(st.msg.GasLimit)
```

Comment:
`st.msg.GasLimit` is set by user and is bigger than the actual `gasused`,  in [refundGas](https://github.com/mantlenetworkio/op-geth/blob/a33e4e11416873c14d868c537afe7e660936ba8f/core/state_transition.go#L519) the remaining gas will refund back to `msg.sender` instead of `sponsor`. is it possible that `msg.sender` steals asset from `sponsor`?